pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -182,9 +182,25 @@ func (c *controller) createInformer(
 	handler := &kube.ChainHandler{}
 	handler.Append(c.notify)
 
+	// Enhance list function with supplied validation function, so that invalid CRD does not enter store.
+	vlf := func(opts meta_v1.ListOptions) (result runtime.Object, err error) {
+		if result, err = lf(opts); err == nil {
+			// This check is primarily needed for tests which use mock config - but helps otherwise also.
+			if obj, ok := result.(crd.IstioObject); ok {
+				err = vf(obj)
+			}
+		}
+		if err != nil {
+			log.Errorf("failed to add CRD while listing. Value: %v, error: %v", result, err)
+			incrementEvent(otype, "addfailure")
```

Comment:
Shouldn’t this be validate failure ?

Reply:
It is generically treated as add failure today as https://github.com/istio/istio/blob/master/pilot/pkg/config/kube/crd/controller/controller.go#L154 - But we can distinguish if it helps?

Reply:
I think I will move it to `k8serrors` - as that is the right stat for it.